periflow/utils/url.py

```diff
@@ -57,3 +15,183 @@ def get_baseurl(url: str) -> str:
     scheme = parsed_url.scheme
     base = parsed_url.netloc
     return f"{scheme}://{base}/"
+
+
+class URLProvider:
+    """Service URL provider."""
+
+    @classmethod
+    def get_auth_uri(cls, path: str) -> str:
+        """Get PFA URI."""
+        raise NotImplementedError  # pragma: no cover
+
+    @classmethod
+    def get_training_uri(cls, path: str) -> str:
+        """Get PFT URI."""
+        raise NotImplementedError  # pragma: no cover
+
+    @classmethod
+    def get_training_ws_uri(cls, path: str) -> str:
+        """Get PFT websocket URI."""
+        raise NotImplementedError  # pragma: no cover
+
+    @classmethod
+    def get_serving_uri(cls, path: str) -> str:
+        """Get PFS URI."""
+        raise NotImplementedError  # pragma: no cover
+
+    @classmethod
+    def get_mr_uri(cls, path: str) -> str:
+        """Get PFR URI."""
+        raise NotImplementedError  # pragma: no cover
+
+    @classmethod
+    def get_meter_uri(cls, path: str) -> str:
+        """Get PFM URI."""
+        raise NotImplementedError  # pragma: no cover
+
+    @classmethod
+    def get_observatory_uri(cls, path: str) -> str:
+        """Get PFO URI."""
+        raise NotImplementedError  # pragma: no cover
+
+
+class ProductionURLProvider(URLProvider):
+    """Production service URL provider."""
+
+    training_url = "https://training.periflow.ai/api/"
+    training_ws_url = "wss://training-ws.periflow.ai/ws/"
+    registry_url = "https://modelregistry.periflow.ai/"
+    serving_url = "https://serving.periflow.ai/"
+    auth_url = "https://auth.periflow.ai/"
+    meter_url = "https://metering.periflow.ai/"
+    observatory_url = "https://observatory.periflow.ai/"
+
+    @classmethod
+    def get_auth_uri(cls, path: str) -> str:
+        """Get PFA URI."""
+        return urljoin(cls.auth_url, path)
+
+    @classmethod
+    def get_training_uri(cls, path: str) -> str:
+        """Get PFT URI."""
+        return urljoin(cls.training_url, path)
+
+    @classmethod
+    def get_training_ws_uri(cls, path: str) -> str:
+        """Get PFT websocket URI."""
+        return urljoin(cls.training_ws_url, path)
+
+    @classmethod
+    def get_serving_uri(cls, path: str) -> str:
+        """Get PFS URI."""
+        return urljoin(cls.serving_url, path)
+
+    @classmethod
+    def get_mr_uri(cls, path: str) -> str:
+        """Get PFR URI."""
+        return urljoin(cls.registry_url, path)
+
+    @classmethod
+    def get_meter_uri(cls, path: str) -> str:
+        """Get PFM URI."""
+        return urljoin(cls.meter_url, path)
+
+    @classmethod
+    def get_observatory_uri(cls, path: str) -> str:
+        """Get PFO URI."""
+        return urljoin(cls.observatory_url, path)
```

Comment:
Is there any reason to implement these methods not in `URLProvider`?
It looks better to move these lines into the parent class, and only URLs are in each children URLProvider.